packages/fast-web-utilities/src/class-names.ts

```diff
@@ -0,0 +1,33 @@
+import { isFunction, isString } from "lodash-es";
+
+type ClassNamesArg = string | (() => string) | [string | (() => string), boolean];
+
+// const isString: (val: unknown) => val is string = (val: unknown): val is string => typeof val === "string"
+
+export function classNames(...args: ClassNamesArg[]): string {
+    let classes: string = "";
+    let leadingChar: string = "";
+    const addLeadingChar: (value: string) => string = (value: string): string =>
+        (leadingChar += value);
+
+    for (let i: number = 0, length: number = args.length; i < length; i++) {
```

Comment:
could `args.length` ever end up as undefined or null?

Reply:
no, it uses spread syntax (or the arguments keyword) so this will always have a `length` property. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax

I'm testing for no arguments also